### PR TITLE
Fix spacing above images and next to scrollbar in media pane

### LINKED ISSUE #1843: Keep spacing under filters and images in media gallery consistent
## Bug Description

If I scroll up, the images should not shift up. 

## Steps to Reproduce

1. GO to the media panel
2. note the size of gap between the first image/video and the filters above "All" "Images" "Video"
3. scroll down the media panel
4. notice the size of gap has shrunk

## Screenshots

Image on far right, you can see that the images are nearly touching the text. 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by diegovar**
## Summary

Fixes the issue where the padding on top of the image list in the media pane was off.
Also ensures that the image carousel is properly centered by executing a js fix via useEffect (as there's no pure CSS solution that I could find)

## Relevant Technical Choices

JS for a layout fix.

## To-do

<none>

## User-facing changes

Fixes the top padding and the carousel being centered.

Before:

before

After:

after

## Testing Instructions

<none>

---

Fixes #1843

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/media/mediaPane.js [hunk 1]
```diff
@@ -252,6 +253,24 @@ function MediaPane(props) {
     [hasMore, isMediaLoading, isMediaLoaded, setNextPage]
   );
 
+  // The media pane is supposed to have a scrollbar that shows on focus and
```
Comment (joannag6) [inline assets/src/edit-story/components/library/panes/media/mediaPane.js]: Maybe separate this comment into:

// TODO(#issue): Implement scrollbar that only appears on MediaPane focus.
// Recalculates padding of Media Pane so it stays centered ... As of 
// May 2020 this cannot be achieved without js (as the scrollbar-gutter prop is no yet ready).

File: assets/src/edit-story/components/library/panes/media/mediaPane.js [hunk 2]
```diff
@@ -252,6 +253,24 @@ function MediaPane(props) {
...
+  // "hovers" over content. As of May 2020 this cannot be achieved without
+  // js (as the scrollbar-gutter prop is no yet ready).
+  useEffect(() => {
```
Comment (joannag6) [inline assets/src/edit-story/components/library/panes/media/mediaPane.js]: Use `useLayoutEffect` instead?

https://kentcdodds.com/blog/useeffect-vs-uselayouteffect

File: assets/src/edit-story/components/library/panes/media/mediaPane.js [hunk 3]
```diff
@@ -252,6 +253,24 @@ function MediaPane(props) {
...
+    const scrollBarWidth =
+      refContainer.current.offsetWidth - refContainer.current.clientWidth;
+    const currentPaddingLeft = parseFloat(
+      window
+        .getComputedStyle(refContainer.current, null)
+        .getPropertyValue('padding-left')
+    );
+    refContainer.current.style['padding-right'] =
+      currentPaddingLeft - scrollBarWidth + 'px';
+  });
```
Comment (joannag6) [inline assets/src/edit-story/components/library/panes/media/mediaPane.js]: This gets called many times, should this have a dependency array?

Maybe something like this: https://medium.com/tech-eatwith/how-to-use-react-refs-with-useeffect-hooks-9a0b31a7b398

**Review by joannag6 — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/components/library/panes/media/mediaPane.js (hunk 2 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/mediaPane.js]: ooh nice!

File: assets/src/edit-story/components/library/panes/media/mediaPane.js (hunk 3 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/mediaPane.js]: Done, I used a callback ref and added the necessary dependencies. Ended up being a bit more verbose.

File: assets/src/edit-story/components/library/panes/media/mediaPane.js [hunk 4]
```diff
@@ -227,13 +228,37 @@ function MediaPane(props) {
 
   const resources = media.filter(filterResource);
 
-  const refContainer = useRef();
+  // TODO(#1698): Ensure scrollbars auto-disappear in MacOS.
+  // Recalculates padding of Media Pane so it stays centered.
```
Comment (joannag6) [inline assets/src/edit-story/components/library/panes/media/mediaPane.js]: Maybe put the "Recalculates ...." part of the comment in the useLayoutEffect func

**Review by joannag6 — APPROVED** [review]
LGTM :)

**Pull request merged**